File: C-Sharp/CodilityScenarios/3. Time Complexity/TapeEquilibrium.cs
using System;
using System.Collections.Generic;
using System.Linq;

namespace CodilityScenarios
{
    /// <summary>
    /// Tape equilibrium.
    /// </summary>
    public class TapeEquilibrium
    {
        public int solution(int[] numericArray)
        {
            // Handle empty array
            if (numericArray.Length == 0)
            {
                return 0;
            }
           
            // Storage for the diff of parts
            List<int> results = new List<int>();
            int interationLimit = numericArray.Length;

            // Iterate through the array and get the diff between i <-- and remaining array values
            for (int i = 1; i < interationLimit; i++)
            {
                results.Add(Math.Abs(numericArray.Take(i).Sum() - numericArray.Skip(i).Sum()));                   
            }

            return results.Min();    
        }
    }
}

File: Python/3. Time Complexity/TapeEquilibrium.py
# Tape equilibrium.
class TapeEquilibrium(object):

    def solution(self, numericArray):

        # defensive code
        if len(numericArray) == 0:
            return 0
    
        # result store
        results = []

        # this sucks, looping and not using the value, also the index modifiers are horrible
        # need to re-write this in a pythonic way
        # essentially get the diff between the value to the left and right of array (n)
        for index, value in enumerate(numericArray[: -1]):
            results.append(abs(sum(numericArray[:index + 1]) - sum(numericArray[index + 1:])))

        # return
        return min(results)